### PR TITLE
[AIP-131] Require google.api.resource_reference on Get methods.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: docs/rules/0131/request-name-reference.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+## Disabling
+
+If you need to violate this rule, use a leading comment above the field.
+Remember to also include an [aip.dev/not-precedent][] comment explaining why.
+
+```proto
+// (-- api-linter: core::0131::request-name-reference=disabled
+//     aip.dev/not-precedent: This is named "book" for historical reasons. --)
+message GetBookRequest {
+  string book = 1 [(google.api.field_behavior) = REQUIRED];
```
Comment (mingzhi) [inline docs/rules/0131/request-name-reference.md]: Wrong example?

File: docs/rules/0131/request-name-reference.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message GetBookRequest {
+  // The `google.api.resource_reference` annotation should also be included.
+  string book = 1 [(google.api.field_behavior) = REQUIRED];
```
Comment (mingzhi) [inline docs/rules/0131/request-name-reference.md]: Wrong example?

File: rules/aip0131/request_name_reference.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var requestNameReference = &lint.FieldRule{
+	Name: lint.NewRuleName(131, "request-name-reference"),
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		return isGetRequestMessage(f.GetOwner())
```
Comment (mingzhi) [inline rules/aip0131/request_name_reference.go]: Do we need to check if the field name is "name"?

File: docs/rules/0131/request-name-reference.md (hunk 1 above)
Comment (lukesneeringer) [reply docs/rules/0131/request-name-reference.md]: Correct example. Both annotations are expected, and it seems more user-friendly to lead toward both in the docs.

File: rules/aip0131/request_name_reference.go (hunk 3 above)
Comment (lukesneeringer) [reply rules/aip0131/request_name_reference.go]: Yes, we do. Thanks. Fixed.

File: docs/rules/0131/request-name-reference.md (hunk 2 above)
Comment (lukesneeringer) [reply docs/rules/0131/request-name-reference.md]: Same as above.

File: rules/aip0131/request_name_reference_test.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestRequestNameReference(t *testing.T) {
+	t.Run("Present", func(t *testing.T) {
+		f := testutils.ParseProto3String(t, `
```
Comment (mingzhi) [inline rules/aip0131/request_name_reference_test.go]: Can you rewrite these two into table tests, like what you did in the other PR? And also can you add a test for a negative case with field name not being "name"? Thank you!

Comment (lukesneeringer) [reply rules/aip0131/request_name_reference_test.go]: I added the not-name check (as a table test with absent). I am declining to write the "present" test as a table test because it gets sufficiently awkward to deal with the template with a multi-line annotation, and I find the existing test more readable.

Comment (mingzhi) [reply rules/aip0131/request_name_reference_test.go]: Fair enough! Thanks!
